### PR TITLE
Enhance cherry-pick-pull.sh to transport also the release notes

### PR DESCRIPTION
**Pull request by vpnachev**
**How to categorize this PR?**

/area dev-productivity
/kind enhancement
/priority normal

**What this PR does / why we need it**:
Enhance cherry-pick-pull.sh to transport also the release notes.

**Which issue(s) this PR fixes**:
Follow up on #3674 

**Special notes for your reviewer**:

**Release note**:

### CODE REVIEW

**Review by rfranzke — DISMISSED** [review]
/lgtm
/invite @ialidzhikov

File: hack/cherry-pick-pull.sh [hunk 1]
```diff
@@ -200,6 +208,10 @@ for pull in "${PULLS[@]}"; do
 
   # remove the patch file from /tmp
   rm -f "/tmp/${pull}.patch"
+
+  # get the release notes
+  notes=$(curl "https://api.github.com/repos/${MAIN_REPO_ORG}/${MAIN_REPO_NAME}/pulls/${pull}" -sS | jq '.body' | grep -Po "\`\`\` *${RELEASE_NOTE_CATEGORY} ${RELEASE_NOTE_TARGET_GROUP}.*?\`\`\`")
```
Comment (ialidzhikov) [inline hack/cherry-pick-pull.sh:213]: On macOs this requires gnu grep.

You can probably improve install-requirements with

```diff
$ git diff
diff --git a/hack/install-requirements.sh b/hack/install-requirements.sh
index 5b5044bff..e673322f3 100755
--- a/hack/install-requirements.sh
+++ b/hack/install-requirements.sh
@@ -45,11 +45,12 @@ Please make sure you have installed the following requirements:
 - GNU Parallel

 Brew command:
-$ brew install coreutils gnu-tar gnu-sed jq parallel
+$ brew install coreutils gnu-tar gnu-sed jq parallel grep

 Please allow them to be used without their "g" prefix:
 $ export PATH=/usr/local/opt/coreutils/libexec/gnubin:\$PATH
 $ export PATH=/usr/local/opt/gnu-tar/libexec/gnubin:\$PATH
 $ export PATH=/usr/local/opt/gnu-sed/libexec/gnubin:\$PATH
+$ export PATH=/usr/local/opt/grep/libexec/gnubin:\$PATH
 EOM
 fi
```

Comment (ialidzhikov) [inline hack/cherry-pick-pull.sh:213]: Hm, just a question. Does this regex always require an empty space between "```" and the release category?

**Review by ialidzhikov — COMMENTED** [review]
Thanks for this PR.
I also noticed that I missed to `chmod +x hack/cherry-pick-pull.sh`. Thanks for catching this as well.

File: hack/cherry-pick-pull.sh (hunk 1 above)
Comment (vpnachev) [reply hack/cherry-pick-pull.sh:213]: I tested it with release notes that have and have not a space between the back ticks and the category, in both cases it was working. The release-note functionality in cc-utils also [have the space](https://github.com/gardener/cc-utils/blob/4df37b6c592329208bc8a0f02f00d9b864aa4f98/github/release_notes/util.py#L414-L419), thus I decided to keep it.

Comment (vpnachev) [reply hack/cherry-pick-pull.sh:213]: Thanks!
I have added it (also sorted the packages alphabetically).

**Review by ialidzhikov — APPROVED** [review]
/lgtm

**Pull request merged**